great_expectations/execution_engine/split_and_sample/sparkdf_data_splitter.py

```diff
@@ -141,7 +143,7 @@ def split_on_date_parts(
         """
         self._validate_date_parts(date_parts)
 
-        date_parts: List[DatePart] = self._convert_date_parts(date_parts)
+        date_parts = self._convert_date_parts(date_parts)
```

Comment:
In this case I would prefer to rename the variable and keep the accurate type hint.

Reply:
The type  of `date_parts` is already declared in the method signature.
```python
date_parts: Union[List[DatePart], List[str]],
```
 It is "narrowed" to `List[DatePart]` at line 146 we can't re-declare types like that.

I don't have a problem with the change you describe but I really try to avoid renaming or making runtime changes as part of these initial type-checkings PRs.
It makes it much cleaner when going through the blame that no runtime changes were made.

Reply:
☝️ If you still feel strongly that you'd like a new variable I can make the change.